src/agent/src/storage/mod.rs

```diff
@@ -55,6 +54,86 @@ pub struct StorageContext<'a> {
     sandbox: &'a Arc<Mutex<Sandbox>>,
 }
 
+/// An implementation of generic storage device.
+pub struct StorageDeviceGeneric {
+    path: String,
+    initialized: bool,
+}
+
+impl std::fmt::Debug for StorageDeviceGeneric {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("StorageDeviceGeneric")
+            .field("path", &self.path)
+            .finish()
+    }
+}
+
+impl Default for StorageDeviceGeneric {
+    fn default() -> Self {
+        StorageDeviceGeneric {
+            path: "".to_string(),
+            initialized: false,
+        }
+    }
+}
+
+impl StorageDeviceGeneric {
+    /// Create a new instance of `StorageStateCommon`.
+    pub fn new(path: String) -> Self {
+        StorageDeviceGeneric {
+            path,
+            initialized: true,
+        }
+    }
+}
+
+impl StorageDevice for StorageDeviceGeneric {
+    fn path(&self) -> &str {
+        &self.path
+    }
+
+    fn cleanup(&self) -> Result<()> {
+        if !self.initialized {
+            return Ok(());
+        } else if self.path.is_empty() {
+            return Err(anyhow!("mountpoint path is empty"));
+        }
+
+        let path = Path::new(&self.path);
+        if !path.exists() {
+            return Ok(());
+        }
+
+        if matches!(is_mounted(&self.path), Ok(true)) {
+            let mounts = vec![self.path.clone()];
+            remove_mounts(&mounts)?;
+        }
+        if matches!(is_mounted(&self.path), Ok(true)) {
+            return Err(anyhow!("failed to umount mountpoint {}", self.path));
+        }
+
+        if path.is_dir() {
+            let is_empty = path.read_dir()?.next().is_none();
+            if !is_empty {
+                return Err(anyhow!("directory is not empty when clean up storage"));
+            }
+            // "remove_dir" will fail if the mount point is backed by a read-only filesystem.
+            // This is the case with the device mapper snapshotter, where we mount the block device
+            // directly at the underlying sandbox path which was provided from the base RO kataShared
+            // path from the host.
+            let _ = fs::remove_dir(path);
+        } else if !path.is_file() {
+            // TODO: should we remove the file for bind mount?
```

Comment:
If the future such as the use of oss, maybe exist bind mount for files.

Reply:
Need to work to ensure that "deleting a data file" is safe:)